README.md

```diff
@@ -70,21 +77,68 @@ window.addEventListener('portalactivate', e => {
 });
 ```
 
-## Use cases
+If `https://example.com/` does not need the predecessor-adoption capability, then it can instead use the generic APIs for reacting to prerendering activation or other relevant changes, outlined [in that explainer](https://github.com/jeremyroman/alternate-loading-modes/blob/gh-pages/browsing-context.md#javascript-api).
 
-_See the ["Key Scenarios"](./key-scenarios.md) document for more detail on each of these, including visualizations._
+### Navigation transitions
+
+As a more realistic example, consider a page which wants to prerender a link and provide a seamless navigation transition when it's clicked. It can do this by hiding the portal until the link is clicked,
+and then animating the portal to full screen before activation.
+
+This might be written like so:
 
-- **Pre-rendering**: by loading a page in a hidden portal, it is possible to pre-render the destination page. Then, activating the portal will instantly display the pre-rendered document in the same browser tab.
+```html
+<a href="https://example.com/" class="seamless">Click me!</a>
+
+<script>
+for (const link of document.querySelectorAll('a.seamless')) {
+  const portal = document.createElement('portal');
+  portal.src = link.href;
+  portal.hidden = true;
+  portal.style = 'position: fixed; top: 0; left: 0; width: 100px; height: 100px;';
+  document.body.append(portal);
 
-  This requires some care and cooperation from both sides to deal with the restrictions on portaled content, and the transition between portaled and activated states.
+  link.onclick = async e => {
+    if (portal.state === 'closed') {
+      // The content couldn't be portaled, likely because it didn't opt-in.
+      // Let the normal link click go through.
+      return;
+    }
+
+    e.preventDefault();
+
+    // Show the portal, and animate it to the whole viewport over 300 milliseconds.
+    portal.hidden = false;
+    if (!matchMedia('(prefers-reduced-motion: reduce)').matches) {
+      await portal.animate([{ width: '100vw', height: '100vh' }], { duration: 300 }).finished;
+    }
```

Comment:
Sorry, one more thing I just noticed. If we have prefers-reduced-motion, it looks like this would flash the portal with its preview size. I think we would want something like:
```
if (!matchMedia('(prefers-reduced-motion: reduce)').matches) {
  portal.hidden = false;
  await portal.animate([{ width: '100vw', height: '100vh' }], { duration: 300 }).finished;
} else {
  portal.style.width = '100vw';
  portal.style.height = '100vh';
  portal.hidden = false;
}
```

Reply:
I assumed activate would be instant (i.e. before the next animation frame), so you wouldn't see the preview. Maybe that's not true in practice?

Reply:
Not sure off the top of my head. In the case where the successor is slow to produce a frame even after activation, we would temporarily show the old content. That might be relevant here.

I was thinking more of the case where the user clicks the link right away so that the activate call below wouldn't be completed immediately while waiting for the initial nav.

And upon further reflection, restyling the preview for the prefers-reduced-motion case is probably unnecessary anyway. We could just activate when ready. That is, move the |portal.hidden = false;| into the if block, so this just behaves like offscreen prerender.

Reply:
I was under the impression that we'd make the portal BC full-viewport and change the URL bar immediately, even if the content was still loading. (Even for the initial nav.)

But yeah, +1 for staying hidden in the prefers-reduced-motion case.

Reply:
Yes, once the navigation matures, the URL bar would immediately reflect the portal context as it still loads. It's just that if the portal context has nothing to show yet, we would keep showing the predecessor's frame (as in image) for a short time to avoid the image flashing. The same thing is done with regular navigations.